src/main/webapp/WEB-INF/view/profile.jsp

```diff
@@ -16,18 +20,29 @@
     <div
       style="width:75%; margin-left:auto; margin-right:auto; margin-top: 50px;">
 
-      <h1>User Profile Pages</h1>
-
       <%
       String currentUser = (String) request.getSession().getAttribute("user");
       String username = (String) request.getAttribute("profilePage");
+      User thisUser = userStore.getUser(username);
 
       if (username.equals("")) { %>
-        <a>User does not exist.</a>
+        <h1>User does not exist.</h1>
       <% } else if (currentUser != null && currentUser.equals(username)) { %>
-        <a>Welcome to your page!</a>
+        <h1>Welcome to your page!</h1>
+        <br/>
+        <a> Edit your bio here! (only you can see this) </a>
+        <form action="/users/<%= currentUser %>" method="POST">
+            <input type="text" name="bio" value="<%= thisUser.getBio() %>" >
+            <br/>
+            <button type="submit">Submit</button>
+        </form>
       <% } else { %>
-        <a>Welcome to <%= username %>'s Page!</a>
+        <h1>Welcome to <%= username %>'s Page!</h1>
+        <br/>
+        <a> About Me </a>
+        <br/>
+        <% String profilePageBio = thisUser.getBio(); %>
+        <a> <%= profilePageBio %> </a>
```

Comment:
I think it would be cool if you could add some comments, I mean the code is self-explainable and pretty well made but maybe adding a few lines would be great for easier reading I think!

Reply:
thanks for letting me know! just added a couple of comments :)